awx/main/credential_plugins/hashivault.py

```diff
@@ -265,6 +265,8 @@ def kv_backend(**kwargs):
 
     if secret_key:
         try:
+            if secret_key != 'data' and 'data' in json['data']:
```

Comment:
now I'm thinking that this is not conservative enough. I think we should add to this `and secret_key not in json['data']`. Because we could be looking up a key, and "data" is another irrelevant key.

Reply:
I think I ended up doing a really liberal one in javascript when I was dealing with Vault because sometimes some frameworks were unwrapping and some weren't, so I had like a recursive-ish deal that unwrapped until there was only one data with a leaf under it.